xrpl4j-core/src/main/java/org/xrpl/xrpl4j/codec/binary/types/AmountType.java

```diff
@@ -142,14 +150,21 @@ public AmountType fromJson(JsonNode value) throws JsonProcessingException {
     if (value.isValueNode()) {
       assertXrpIsValid(value.asText());
 
-      UnsignedByteArray number = UnsignedByteArray.fromHex(
+      final boolean isValueNegative = value.asText().startsWith("-");
+      final UnsignedByteArray number = UnsignedByteArray.fromHex(
         ByteUtils.padded(
-          UnsignedLong.valueOf(value.asText()).toString(16),
-          64 / 4
+          UnsignedLong
+            .valueOf(isValueNegative ? value.asText().substring(1) : value.asText())
+            .toString(16),
+          16 // <-- 64 / 16
         )
       );
-      byte[] rawBytes = number.toByteArray();
-      rawBytes[0] |= 0x40;
+      final byte[] rawBytes = number.toByteArray();
+      if (isValueNegative) {
+        rawBytes[0] |= 0x00;
```

Comment:
Isn't `rawBytes[0] |= 0x00` the same as just `rawBytes[0]`? Wondering if we should just do 
```
if (!isValueNegative) {
  rawBytes[0] |= 0x40;
}
```

Reply:
Great catch! Updated.